crates/ruff_linter/src/noqa.rs

```diff
@@ -38,7 +39,16 @@ pub fn generate_noqa_edits(
     let file_directives = FileNoqaDirectives::extract(locator, comment_ranges, external, path);
     let exemption = FileExemption::from(&file_directives);
     let directives = NoqaDirectives::from_commented_ranges(comment_ranges, external, path, locator);
-    let comments = find_noqa_comments(diagnostics, locator, &exemption, &directives, noqa_line_for);
+    // This is called by ruff_server, which already filtered diagnostics via check_path
```

Comment:
We also call this function from our tests. Would it make sense to lift `Suppressions` out of `generate_noqa_edits` and instead require the caller to pass the suppressions?